Libraries/Animated/src/nodes/AnimatedNode.js

```diff
@@ -36,11 +43,99 @@ class AnimatedNode {
   /* Methods and props used by native Animated impl */
   __isNative: boolean;
   __nativeTag: ?number;
+
+  constructor() {
+    this._listeners = {};
+  }
+
   __makeNative() {
     if (!this.__isNative) {
       throw new Error('This node cannot be made a "native" animated node');
     }
+
+    if (Object.keys(this._listeners).length) {
+      this._startListeningToNativeValueUpdates();
+    }
+  }
+
+  /**
+   * Adds an asynchronous listener to the value so you can observe updates from
+   * animations.  This is useful because there is no way to
+   * synchronously read the value because it might be driven natively.
+   *
+   * See http://facebook.github.io/react-native/docs/animatedvalue.html#addlistener
+   */
+  addListener(callback: any): string {
```

Comment:
I know it's not perfect to use `any`, but we need to override this type in `AnimatedValueXY` with another type which is related to another logic. 😭

Reply:
@osdnk maybe we can use `callback: (value: any) => mixed` instead?

Reply:
`callback: (value: any) => void`

Reply:
@osdnk using `mixed` or `any` is better in this case. `void` will prevent async functions to be used for the listener: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swBDAEwoDU5CKAZQgZ3wFNs2AnAClzJgwARmVwBrAFxgeANwGY2UstgCeASjABeAHxgZdChoDeAX3RY8REuSoBZQgA82jFu069+gkeKmz5i8lUNHTAAW0dnYzNUShoDJlYObh4yZhU8aWDdI1QwMGBgMAo4MGY4ULZ8AAtCbABzVBM1AG50WPsnF0T3FLSMniywHLyCopKyiurahqbmoA

Reply:
Okay, well, I didn't want to do it previously, cause it changes existing code